framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1626,6 +1626,40 @@ public Void visitExecutable(AnnotatedExecutableType type, AnnotatedTypeFactory p
             // Also skip constructor return types (which somewhat act like
             // the receiver).
             MethodSymbol methodElt = (MethodSymbol) type.getElement();
+
+            // If a constructor declaration is not explicitly annotated,
+            // annotate it with the type on the class declaration.
+            Element elem = type.getElement();
+            if (elem.getKind() == ElementKind.CONSTRUCTOR) {
+                AnnotatedDeclaredType returnType = (AnnotatedDeclaredType) type.getReturnType();
+                if (returnType.getAnnotations().isEmpty()) {
```

Comment:
What is the purpose of this check?  Is it just an optimization to avoid a later check, or is it necessary for correctness?
If defaults are already applied, might this test return those defaults and thus incorrectly avoid executing the body?  That is, I could imagine that if defaults are already applied, `getAnnotations()` might be non-empty even though the user did not write any annotations explicitly.  A word of explanation would be useful.

Reply:
I have added more explanation in my latest commit.